### PR TITLE
Final edit for DG

### LINKED ISSUE #291: Final edits to DG
Finish editing DG

### PR DESCRIPTION
**Pull request by mingyuanc**
Fixes #291

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1012,7 +1101,7 @@ otherwise)
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1101]: Should change to 1e1. ProfBook creates ToDo for either all students..... (Not Deadline)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
+    - Prerequisites: There exist a group with GroupId `grp-001` with at least one student.<br>
+      Prerequisites: At least one of the student does not have the todo task, Assignment 1.<br>
       Test case: `todo ~/grp-001 --desc Assignment 1 --all allStu`<br>
-      Expected: The todo task `Assignment 1` will be allocated all students in `grp-001`.
+      Expected: The todo task `Assignment 1` will be allocated all students in `grp-001` that does not already have the todo task.
 
-    - Prerequisites: There is at least one student under the root directory and not all the student have the todo task.
-      Test case: `todo ~ --desc Assignment 1 --all allStu`<br>
-      Expected: The todo task `Assignment 1` will be allocated to all students under root directory.
+    - Prerequisites: There exist at least one student in ProfBook and at least of the students does not have the todo task, Assignment 1.<br>
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1413]: missing a word 'one' after 'at least'.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
+    - Prerequisites: There exist a group with GroupId `grp-001` with at least one student.<br>
+      Prerequisites: At least one of the student does not have the deadline task, Assignment 1.<br>
       Test case: `deadline ~/grp-001 --desc Assignment 1 --datetime 2023-10-11 23:59 --all allStu`<br>
-      Expected: The deadline task `Assignment 1` will be allocated all students in group `grp-001`.
+      Expected: The deadline task `Assignment 1` will be allocated all students in `grp-001` that does not already have the task.
 
-    - Prerequisites: There is at least one student under the root directory and not all the student have the deadline task.
-      Test case: `deadline ~ --desc Assignment 1 --datetime 2023-10-11 23:59 --all allStu`<br>
-      Expected: The deadline task `Assignment 1` will be allocated to all students under root directory.
+    - Prerequisites: There exist at least one student in ProfBook and at least of the students does not have the deadline task, Assignment 1.<br>
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1460]: missing a word 'one' after 'at least'.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
     - Test case: `mark a`<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
-
-    - Test case: `mark x` (where x is not a number)<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
+      Expected: No task would be marked, an error will be shown.
 
-    - Other incorrect `mark` commands to try: `mark`<br>
-      Expected: An error message indicating the command format is invalid will be displayed.
+    - Other incorrect `mark` commands to try: `mark -2`, `mark x` (Where x is positive integer which is less than current task size)<br>
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1493]: Should be 'greater than the size of current task list'.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: An error message indicating the command format is invalid will be displayed.
 
-    - Test case: `unmark a` (where x is not a number)<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
+    - Test case: `unmark a`<br>
+      Expected: No task would be un-marked, an error will be shown.
 
-    - Test case: `unmark x` (where x is not a number)<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
+    - Other incorrect `un-mark` commands to try: `unmark -2`, `unmark x` (Where x is positive integer which is less than current task size)<br>
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1513]: same as above.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: No task would be marked, an error message indicating that the index given is invalid.
 
-    - Other incorrect `unmark` commands to try: `unmark`<br>
-      Expected: An error message indicating the command format is invalid will be displayed.
+### Delete a task
 
-### Delete task
+- Deletes a specified task as done for a student or group,
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1518]: Perhaps remove the word 'as done' will make it more clearer.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
     - Test case: `rmt a`<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
-
-    - Test case: `rmt x` (where x is not a number)<br>
-      Expected: An error message indicating the index provided is not a non-zero unsigned integer will be displayed.
+      Expected: No task would be deleted, an error will be shown.
 
-    - Other incorrect `rmt` commands to try: `rmt`<br>
-      Expected: An error message indicating the command format is invalid will be displayed.
+    - Other incorrect `rmt` commands to try: `rmt -2`, `rmt x` (Where x is positive integer which is less than current task size)<br>
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1533]: same as above. Should be 'greater than the size of current task list'.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: `profbook.json` is updated with `Tejas removed`. Sample output [here](https://github.com/AY2324S1-CS2103T-W15-2/tp/tree/master/docs/sample/removeTejas.json).
 
 - Clearing the save file.
 
-    - Prerequisites: Have profbook.json in the data folder. Perform the previous step if the file isn’t there.
+    - Prerequisites: Have `profbook.json` in the data folder. Perform the previous step if the file isn’t there.
 
     - Test case: `clear`
-      Expected: An empty profbook.json file like [here](https://github.com/AY2324S1-CS2103T-W15-2/tp/tree/master/docs/sample/empty.json)).
+      Expected: An empty `profbook.json` file like [here](https://github.com/AY2324S1-CS2103T-W15-2/tp/tree/master/docs/sample/empty.json)).
```
Comment (NgChunMan) [inline docs/DeveloperGuide.md:1606]: There is a redundant ')' bracket at the end of the link.

**Review by NgChunMan — COMMENTED** [review]
Here are the changes that I noticed while proof reading the DG.

**Review by Nid21cs — CHANGES_REQUESTED** [review]

**Review by NgChunMan — CHANGES_REQUESTED** [review]
Just a few minor change required.

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1199,371 +1301,311 @@ testers are expected to do more *exploratory* testing.
...
-
 ### Saving data
 
 - Getting the default save file.
 
     - Prerequisites: Place ProfBook.jar in an empty home folder. Perform the following step in the root directory.
 
     - Test case: `todo grp-001/0001Y --desc Assignment One`
-      Expected: profbook.json appears in data folder inside home folder. Student with name `Tejas` has a field history in profbook.json whereas the other students do not. Sample output [here](https://github.com/AY2324S1-CS2103T-W15-2/tp/tree/master/docs/sample/addTodo.json).
+      Expected: `profbook.json` appears in data folder inside home folder. Student with name `Tejas` has a field history in profbook.json whereas the other students do not. Sample output [here](https://github.com/AY2324S1-CS2103T-W15-2/tp/tree/master/docs/sample/addTodo.json).
```
Comment (zannloo) [inline docs/DeveloperGuide.md:1596]: Expected does not start on a new line in the pdf

**Review by zannloo — COMMENTED** [review]
I am unable to comment on the specific line of code for some so here are the rest of the comments:

Extension explanation for ls and cat is different when path not specified, would be good to standardise but the current one is clear also. 

For use case: add a student, under extension, path provided can be incorrect as well, not sure if ‘given parameters’ include path. 

For use case: edit a group’s details, under extension, user can choose to not input a path if they are already in the desired group directory.

For use case: Delete a task, under mss would it be better to standardise the use of task index instead of using specific id? Also, phrasing of 1a is a little awkward, maybe omit out the `cat` and rephrase a little. 

For use case: Create a group, under extensions 1b, would it be clearer to say group id instead?

For use case: create todo, allStu can add deadline to all student under profbook as well. -> same with deadlines

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -382,79 +426,89 @@ Below is an activity diagram showing the general activity of the add student com
...
+<div style="page-break-after: always;"></div>
 
 ### Creating a task
 
 #### Implementation
 
 Creating and adding a task is one of the key feature of ProfBook. Currently, we support two types of tasks,
-namely `ToDo` and `Deadline` Tasks. Both this tasks extends from the abstract `Task` class which add to its
+namely `ToDo` and `Deadline` tasks. Both this tasks extends from the abstract `Task` class which add to its
 extensibility. It is important to note that currently, you can only add tasks to Group and Students. Needless to say,
```
[the remark this answers is not in the document]
Comment (mingyuanc) [reply docs/DeveloperGuide.md:447]: should be can only add tasks to groups and students

**Review by zannloo — APPROVED** [review]
LGTM

**Review by NgChunMan — DISMISSED** [review]
LGTM.

**Review by Nid21cs — APPROVED** [review]

**Pull request merged**